### PR TITLE
docs(self-hosted configuration): update binarySource

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Update `binarySource` docs
  - Drop list of _Supported tools for dynamic install_
  - Replace word `binary`/`binaries` with `tool`/`tools`

## Context

Inspired by a Slack comment by @viceice, where they proposed to remove the long list.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/self-hosted-configuration.md [hunk 1]
```diff
@@ -150,46 +150,21 @@ If the "development branch" is configured but the branch itself does not exist (
...
-- `yarn`
-
-If all projects are managed by Hermit, you can tell Renovate to use the tooling versions specified in each project via Hermit by setting `binarySource=hermit`.
-
-Tools not on this list fall back to `binarySource=global`.
+This mode means that Renovate will dynamically install the version of binaries available, if supported. // I think we now support install-mode for all binaries/tools???
+
+If all projects are managed by Hermit, you can tell Renovate to use the binary versions specified in each project via Hermit by setting `binarySource=hermit`.
+
+Binaries not on this list fall back to `binarySource=global`.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:167]: We don't have a list any more, so what should the sentence be now?

File: docs/usage/self-hosted-configuration.md [hunk 2]
```diff
@@ -150,46 +150,21 @@ If the "development branch" is configured but the branch itself does not exist (
...
-- `pnpm`
-- `poetry`
-- `python`
-- `rust`
-- `yarn`
-
-If all projects are managed by Hermit, you can tell Renovate to use the tooling versions specified in each project via Hermit by setting `binarySource=hermit`.
-
-Tools not on this list fall back to `binarySource=global`.
+This mode means that Renovate will dynamically install the version of binaries available, if supported. // I think we now support install-mode for all binaries/tools???
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:163]: This line needs a suggestion from the maintainers. 😉

File: docs/usage/self-hosted-configuration.md [hunk 3]
```diff
@@ -150,46 +150,21 @@ If the "development branch" is configured but the branch itself does not exist (
...
+
+If all projects are managed by Hermit, you can tell Renovate to use the binary versions specified in each project via Hermit by setting `binarySource=hermit`.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:165]: The Hermit sentence feels really clunky.

File: docs/usage/self-hosted-configuration.md [hunk 4]
```diff
@@ -150,46 +150,21 @@ If the "development branch" is configured but the branch itself does not exist (
 ## binarySource
 
 Renovate often needs to use third-party binaries in its PRs, like `npm` to update `package-lock.json` or `go` to update `go.sum`.
-By default, Renovate uses a child process to run such tools, so they must be:
+By default, Renovate uses a child process to run such binaries, so they must be:
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:153]: We need to decide if we want to use `binaries` or `tools` in the docs. Right now we use `tools` and `binaries`, which is a bit confusing. 😄

File: docs/usage/self-hosted-configuration.md [hunk 5]
```diff
@@ -160,34 +160,9 @@ For this to work, `docker` needs to be installed and the Docker socket available
...
-- `python`
-- `rust`
-- `yarn`
-
-If all projects are managed by Hermit, you can tell Renovate to use the tooling versions specified in each project via Hermit by setting `binarySource=hermit`.
+This mode means that Renovate will dynamically install the version of tools available, if supported. // I think we now support install-mode for all tools???
+
+If all projects are managed by Hermit, you can tell Renovate to use the tool versions specified in each project via Hermit by setting `binarySource=hermit`.
 
 Tools not on this list fall back to `binarySource=global`.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:167]: We don't have a list any more, so what should the sentence be now?

File: docs/usage/self-hosted-configuration.md [hunk 6]
```diff
@@ -160,34 +160,9 @@ For this to work, `docker` needs to be installed and the Docker socket available
...
-- `pip-compile`
-- `pipenv`
-- `pnpm`
-- `poetry`
-- `python`
-- `rust`
-- `yarn`
-
-If all projects are managed by Hermit, you can tell Renovate to use the tooling versions specified in each project via Hermit by setting `binarySource=hermit`.
+This mode means that Renovate will dynamically install the version of tools available, if supported. // I think we now support install-mode for all tools???
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:163]: Can you suggest a change to this line, I don't know how to change it now we support `install-mode` for all tools? 😕

File: docs/usage/self-hosted-configuration.md [hunk 7]
```diff
@@ -160,34 +160,9 @@ For this to work, `docker` needs to be installed and the Docker socket available
...
+
+If all projects are managed by Hermit, you can tell Renovate to use the tool versions specified in each project via Hermit by setting `binarySource=hermit`.
```
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:165]: The Hermit sentence feels clunky, and a bit out of place.

Also I would expect to see an example of the correct Hermit configuration, maybe?

**Review by HonkingGoose — COMMENTED** [review]
Copy/pasting my review comments onto the changed file, so the maintainers can fix them with suggestions/comments. 😉

File: docs/usage/self-hosted-configuration.md (hunk 6 above)
Comment (rarkins) [inline docs/usage/self-hosted-configuration.md:163]:
```suggestion
This mode means that Renovate will dynamically install the desired version of each tool needed.
```

File: docs/usage/self-hosted-configuration.md (hunk 7 above)
Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:165]: I'm ok to leave this as-is

File: docs/usage/self-hosted-configuration.md (hunk 5 above)
Comment (rarkins) [inline docs/usage/self-hosted-configuration.md:167]:
```suggestion
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**